collada/source.py

```diff
@@ -126,7 +126,10 @@ def __init__(self, id, data, components, xmlnode=None):
         """The unique string identifier for the source"""
         self.data = data
         """Numpy array with the source values. This will be shaped as ``(-1,N)`` where ``N = len(self.components)``"""
-        self.data.shape = (-1, len(components) )
+        try:
+            self.data.shape = (-1, len(components) )
+        except BaseException:
```

Comment:
which numpy exception is thrown here? any way to use a more specific exception type instead of broadly catching BaseException?

Reply:
It's a `ValueError` when the reshape fails (i.e. from (1,) to (-1, 3)) or something.

I just changed it to a conditional check which should be a little narrower and more explicit. I also added `test_ignore.py` and the 2 models from `assimp`'s BSD-licensed models that hit edge cases.

Reply:
perfect, thank you!